src/merkle/mmr/accumulator.rs

```diff
@@ -59,3 +59,13 @@ impl MmrPeaks {
             .verify(opening.relative_pos() as u64, value, root)
     }
 }
+
+impl From<&MmrPeaks> for Vec<Felt> {
+    fn from(mmr: &MmrPeaks) -> Self {
+        let mut elements = Vec::with_capacity(4 * mmr.peaks.len() + 2);
+        elements[0] = Felt::new(mmr.num_leaves as u64);
+        elements[1] = Felt::new(mmr.peaks.len() as u64);
```

Comment:
This shouldn't be necessary, the number of peaks is equal to the `popcount` onf the `num_leaves`

Reply:
I think including just the number of leaves could be interesting - e.g., something like:
```Rust
fn from(mmr: &MmrPeaks) -> Self {
     let mut elements = Vec::with_capacity(4 * mmr.peaks.len() + 1);
     elements[0] = Felt::new(mmr.num_leaves as u64);
     elements[1..].copy_from_slice(&mmr.peaks.as_slice().concat());
    elements
}
```
But we don't currently have a good way to write non-word aligned data back into the advice provider (see https://github.com/0xPolygonMiden/miden-vm/pull/834#pullrequestreview-1379950440).

Reply:
I think we should add the padding data, without it the masm code would crash

Reply:
good catch.

Reply:
I made some changes but I will need to test this properly.

Reply:
FYI: https://github.com/0xPolygonMiden/miden-vm/pull/841/files#r1164066428